src/makeInfinityAdder.js

```diff
@@ -6,7 +6,29 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  return function adder(num) {
+    if (typeof num === 'undefined') {
+      return 0;
+    }
+
+    let sum = 0;
+
+    sum += num;
+
+    return function newAdder(newNum) {
+      if (typeof newNum === 'undefined') {
+        const result = sum;
+
+        sum = 0;
+
+        return result;
+      }
+
+      sum += newNum;
+
+      return newAdder;
+    };
+  };
```

Comment:
Actually, you don't need 2 nested functions in this task. Let's try to implement with only one function.